app/models/press_notice.rb

```diff
@@ -66,4 +68,11 @@ def audit_press_notice!
   def joined_reasons
     reasons.values.join(", ")
   end
+
+  def update_consultation_end_date!
+    return unless consultation
+    return unless saved_changes.include? "published_at"
+
+    consultation.update!(end_date: published_at + 21.days)
```

Comment:
I'm just thinking - for the site notice we give some extra wiggle room and add a couple more days, in case the applicant prints it and then doesn't put it up until the next day. 

In which case, it could be possible that the consultation date might be 23 days ahead, and here we're reducing it by two days or so. 

Could we check whether this date is less or more than the current date, and then set it?

Reply:
Somewhere we do "next business day + 21 days", for example?